src/metrics/elasticsearch_metric.ts

```diff
@@ -0,0 +1,64 @@
+import { AbstractMetric, Datasource, MetricId, MetricQuery, MetricResults  } from './metric';
+
+import * as _ from 'lodash';
+
+
+export class ElasticsearchMetric extends AbstractMetric {
+  constructor(datasource: Datasource, targets: any[], id?: MetricId) {
+    super(datasource, targets, id);
+  }
+
+  getQuery(from: number, to: number, limit: number, offset: number): MetricQuery {
+    let data = this.datasource.data.split('\n').map(d => d !== '' ? JSON.parse(d) : d);
+
+    data[1].size = limit;
+    data[1].from = offset;
+
+    let range = data[1].query.bool.filter.filter(f => _.has(f, 'range'))[0].range;
+    range['@timestamp'].gte = from.toString();
+    range['@timestamp'].lte = to.toString();
+    data = data.map(d => JSON.stringify(d)).join('\n');
+
+    return {
+      url: this.datasource.url,
+      method: 'POST',
+      schema: {
+        data: data
+      }
+    }
+  }  
+
+  getResults(res): MetricResults {
+    let emptyResult = {
```

Comment:
you use `emptyResult ` only in `if` below. I think we can create this object only if `if` is *if*